rules/do-you-use-loggermessage-in-net.md/rule.md

```diff
@@ -0,0 +1,118 @@
+---
+seoDescription: Learn how to optimize logging in .NET applications using the LoggerMessage class. Discover the performance benefits, such as reduced object allocations and efficient message parsing, and get practical tips on how to implement this pattern for better logging efficiency and structured logs in high-performance scenarios.
+type: rule
+title: Do you use LoggerMessage in .NET?
+uri: do-you-use-loggermessage-in-net
+authors:
+  - title: Tom Iwainski
+    url: https://www.ssw.com.au/people/thomas-iwainski
+related: []
+created: 2024-11-05T12:52:08.000Z
+archivedreason: null
+guid: 2feb782f-0cee-4022-95d9-f44b9f739b90
+---
+
+Logging is a critical component in modern applications, but it can easily introduce performance overhead.
+
+The LoggerMessage class in .NET offers a solution to reduce the performance costs associated with logging. By utilizing this pattern, developers can optimize logging with fewer object allocations and reduced overhead.
```

Comment:
This article should not be able the `LoggerMessage` class, but instead about Compile-time logging source generation.

Reply:
not sure if i understand your message, can you paraphrase or explain what you mean?

Reply:
Updated. TLDR; We should talk more about the source generated logs driven by the `[LoggerMessage]` attribute.